codechain/config/mod.rs

```diff
@@ -395,10 +395,9 @@ impl Mining {
     }
 
     pub fn overwrite_with(&mut self, matches: &clap::ArgMatches) -> Result<(), String> {
-        if matches.is_present("no-miner") {
-            self.disable = Some(true);
+        if matches.is_present("author") || matches.is_present("engine-signer") {
```

Comment:
You should check self.author and self.engine_signer also.
I think it's better to check at the last of the method.

Reply:
Ok. I moved the checking to the last part and made to check self.author and self.engine_signer